pkg/apis/autoscaling/validation/validation.go

```diff
@@ -36,6 +37,33 @@ const (
 	MaxStabilizationWindowSeconds int32 = 3600
 )
 
+// HPAValidationOptions contains the different settings for HPA validation
+type HPAValidationOptions struct {
+	AllowMismatchedTargetTypeAndValue bool
```

Comment:
since the only thing we're ratcheting here is objectMetrics, make the name explicitly about that metric type:

```suggestion
	AllowMismatchedObjectMetricTypeAndValue bool
```